src/renderer/components/+namespaces/namespace.store.ts

```diff
@@ -71,24 +73,11 @@ export class NamespaceStore extends KubeObjectStore<Namespace> {
     });
   }
 
-  @computed
   private get initialNamespaces(): string[] {
-    const namespaces = new Set(this.allowedNamespaces);
-    const prevSelectedNamespaces = storage.get();
+    const namespaces = this.allowedNamespaces;
 
-    // return previously saved namespaces from local-storage (if any)
-    if (prevSelectedNamespaces) {
-      return prevSelectedNamespaces.filter(namespace => namespaces.has(namespace));
-    }
-
-    // otherwise select "default" or first allowed namespace
-    if (namespaces.has("default")) {
-      return ["default"];
-    } else if (namespaces.size) {
-      return [Array.from(namespaces)[0]];
-    }
-
-    return [];
+    // return all namespaces (empty list) if "default" namespace doesn't exist
+    return selectedNamespaces.get().filter(namespace => namespaces.includes(namespace));
```

Comment:
I agree with @Nokel81 , we should try to keep the existing behaviour here.

Reply:
It's only for initial first run. Otherwise picking the first namespace also not the best solution..

Reply:
Why, I think it is a better default than loading everything if `"default"` isn't there.

Reply:
Still we shouldn't change this behaviour in a patch release. There is a reason why we don't default to all namespaces -> it might easily burn clusters 🔥 .